### PR TITLE
[READY] Mafia Documentation and Improvements

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Documents the controller datum AND...

### improves these parts of mafia:
* You can no longer suicide while playing mafia (just ghost)
* Removes 9 assistant 3 changeling setup, it's intensely boring and with no investigatory roles the game resorts to random guessing
* Corrects one of the setups to actually have 12 players, this was a really, really bad issue trust me. less than the required amount and most of the time you will only play the lower pop setup, more than the required amount and there aren't enough landmarks breaking EVERYTHING EVER
* Starting noise on day 1 to tell people they got in a game
* TRAITORS CAN ACTUALLY WIN OOPS
* Fugitive protection text starts at sundown and not the start of resolve
* Psychologist gets clearer text about their power
* Being signed up or part of an active game will now send you messages about the ongoing game.
* Grammar

## Why It's Good For The Game

Mafia got prematurely merged but that's alright, just finishing my work here

And God said, “Let there be documentation,” and there was documentation.

image

## Changelog
:cl:
del: Removes the 9 assistants 3 changeling setup, yech.
fix: Lotta bugfixes for mafia!
soundadd: Mafia now alerts you when starting a game!
spellcheck: grammar + better text for some roles at better times of the game
code: LET THERE BE MAFIA DOCUMENTATION!
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
Needs the documentation tag too 😎

**Review by bobbah — COMMENTED** [review]
Can you go through your proc documentation and make sure you are documenting the arguments properly? It looks like in some cases you described the arguments in passing, but you should be explicitly listing the arguments and what they are for.

[You can refer to the documentation requirements here for more information.](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md)

**Comment by bobbahbrown** [comment]
lookin gr8

File: code/modules/mafia/controller.dm [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
+	///like the lynch period, players need to see what the other players in the game's roles were
+	var/victory_lap_period = 20 SECONDS
+
+	///template picked when the game starts. used for the name and desc reading
 	var/datum/map_template/mafia/current_map
+	///map generation tool that deletes the current map after the game finishes
 	var/datum/map_generator/massdelete/map_deleter
 
-	var/list/current_setup_text //Redable list of roles in current game
+	///Readable list of roles in current game (Psychologist x1, Clown x2, etc)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mafia/controller.dm:37]:
```suggestion
	///Readable list of roles in current game (Psychologist = 1, Clown = 2, etc)
```
??

File: code/modules/mafia/controller.dm [hunk 2]
```diff
@@ -1,35 +1,62 @@
...
-	var/player_outfit = /datum/outfit/mafia //todo some fluffy outfit
+	///starting outfit for all mafia players. it's just a grey jumpsuit.
+	var/player_outfit = /datum/outfit/mafia
 
-	var/list/landmarks = list() //spawn points for players, each one has a house
-	var/town_center_landmark = null //town center for when people get put on trial
+	///spawn points for players, each one has a house
+	var/list/landmarks = list()
+	///town center for when people get put on trial
+	var/town_center_landmark = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mafia/controller.dm:46]:
```suggestion
	var/town_center_landmark
```

File: code/modules/mafia/controller.dm [hunk 3]
```diff
@@ -1,35 +1,62 @@
...
 
-	var/list/votes = list() //group voting on one person, like putting people to trial or choosing who to kill as mafia
-	//and these are the judgement phase votes, aka people sorting themselves into guilty and innocent lists. whichever has more wins!
+	///group voting on one person, like putting people to trial or choosing who to kill as mafia
+	var/list/votes = list()
+	///and these (judgement_innocent_votes and judgement_guilty_votes) are the judgement phase votes, aka people sorting themselves into guilty and innocent lists. whichever has more wins!
 	var/list/judgement_innocent_votes = list()
 	var/list/judgement_guilty_votes = list()
+	///current role on trial for the judgement phase, will die if guilty is greater than innocent
 	var/datum/mafia_role/on_trial = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mafia/controller.dm:54]:
```suggestion
	var/datum/mafia_role/on_trial
```

File: code/modules/mafia/controller.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/mafia/controller.dm:37]: This is literally what is being sent, a list of "Psychologist x1", i'll put quotes to make it clearer

**Pull request merged**